pyanaconda/modules/common/versions.py

```diff
@@ -0,0 +1,16 @@
+from functools import cmp_to_key
+
+import rpm
+
+
+__all__ = ['rpm_version_key']
+
+
+def rpm_version_cmp(v1, v2):
+    # TODO do we need to parse epoch? nothing in the tests indicates that
+    v1, _, r1 = v1.partition('-')
+    v2, _, r2 = v2.partition('-')
+    return rpm.labelCompare(('0', v1, r1), ('0', v2, r2))
```

Comment:
I was playing around with the `rpm.labelCompare` function and it looks like it is able to handle strings. Therefore, it is not necessary to parse them and we can create the key function directly from `rpm.labelCompare`.

Reply:
WOW. Indeed. I think it wasn't always the case, but I've checked with RPM 4.17 on Fedora 36 and it even works there.

Reply:
Bloody hell that makes the code super simple.

Reply:
For clarity.

### RHEL 8

```pycon
>>> import rpm
>>> rpm.labelCompare("1", "2")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: argument 1 must be sequence of length 3, not 1
```

### RHEL 9

```pycon
>>> import rpm
>>> rpm.labelCompare("1", "2")
-1
```

Reply:
https://github.com/rpm-software-management/rpm/pull/1221